src/UI/SignTransactionsModals/SignWithWalletConnectModal/index.tsx

```diff
@@ -35,14 +35,23 @@ export const SignWithWalletConnectModal = ({
       safeRedirect(callbackRoute);
     }
   };
+
+  const { show, hide } = useDappModal({
+    modalDialogClassName: classes.wrapper
+  });
+
+  useEffect(() => {
```

Comment:
This tends to get repeated a lot, maybe we can move the show condition in params for useDappModal and move the useEffect inside the hook